crypto/signer.go

```diff
@@ -24,7 +24,7 @@ type Signer struct {
 }
 
 // NewSigner returns a new signer.
-func NewSigner(signer crypto.Signer, _ crypto.Hash) *Signer {
+func NewSigner(signer crypto.Signer) *Signer {
```

Comment:
It seems like this whole struct is now removable.

Reply:
Yes, it could be just replaced, but keeping the type allows for controlling what methods are available (only `Public()` now!), which helps reasoning about it for later PRs.

The last usage of it (and there's only *one*!) is intertwined with a lot of other things, more PRs are coming to untangle this in a way that can be understood and reviewed more comfortably.

Reply:
SGTM. Btw, if the purpose is to reduce the number of available methods, a shorter approach might be introducing an interface with just those methods.